protocols/luci-proto-bonding/htdocs/luci-static/resources/protocol/bonding.js

```diff
@@ -0,0 +1,442 @@
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var o;
+
+		o = s.taboption('general', form.Value, 'ipaddr',
+				_('IPv4 address'),
+				_('The local IPv4 address'));
+		o.datatype = 'ip4addr';
+		o.validate = validateEmpty;
+
+		o = s.taboption('general', form.Value, 'netmask',
+				_('IPv4 netmask'),
+				_('The local IPv4 netmask'));
+		o.datatype = 'ip4addr';
+		o.validate = validateEmpty;
+		o.value("255.255.255.0");
+		o.value("255.255.0.0");
+		o.value("255.0.0.0");
+
+		o = s.taboption('advanced', form.MultiValue, 'slaves',
+				_('Slave Interfaces'),
+				_('Specifies which slave interfaces should be attached to this bonding interface'));
+
+		o.load = function(section_id) {
+			return getSelectableSlaves(section_id).then(L.bind(function(devices) {
+				for (var i = 0; i < devices.length; i++) {
+					this.value(devices[i], devices[i]);
+				}
+
+				if (devices.length == 0) {
+					this.placeholder = _('No more slaves available, can not save interface');
+					this.value('', '');
+					return '';
+				}
+
+				var if_slaves = uci.get('network', section_id, 'slaves');
+
+				if (if_slaves != null) {
+					return if_slaves;
+				}
+
+				return;
+			}, this));
+		};
+		o.cfgvalue = function(section_id) {
+			return uci.get('network', section_id, 'slaves');
+		};
+		o.validate = validateEmpty;
+		
+		o = s.taboption('advanced', form.ListValue, 'bonding_policy',
+				_('Bonding Policy'),
+				_('Specifies the mode to be used for this bonding interface'));
+		o.default = 'balance-rr';
+		o.value('balance-rr', _('Round-Robin policy (balance-rr, 0)'));
+		o.value('active-backup', _('Active-Backup policy (active-backup, 1)'));
+		o.value('balance-xor', _('XOR policy (balance-xor, 2)'));
+		o.value('broadcast', _('Broadcast policy (broadcast, 3)'));
+		o.value('802.3ad', _('IEEE 802.3ad Dynamic link aggregation (802.3ad, 4)'));
+		o.value('balance-tlb', _('Adaptive transmit load balancing (balance-tlb, 5)'));
+		o.value('balance-alb', _('Adaptive load balancing (balance-alb, 6)'));
+
+		o = s.taboption('advanced', form.ListValue, 'primary',
+				_('Primary Slave'),
+				_('Specifies which slave is the primary device. It will always be the active slave while it is available'));
+		o.depends('bonding_policy', 'active-backup');
+		o.depends('bonding_policy', 'balance-tlb');
+		o.depends('bonding_policy', 'balance-alb');
+
+		o.load = function(section_id) {
+			return getSelectableSlaves(section_id).then(L.bind(function(devices) {
+				for (var i = 0; i < devices.length; i++) {
+					this.value(devices[i], devices[i]);
+				}
+
+				if (devices.length == 0) {
+					this.placeholder = _('No more primaries available, can not save interface');
+					this.value('', '');
+					return '';
+				}
+
+				var primary = uci.get('network', section_id, 'primary');
+
+				if (primary != null) {
+					return primary;
+				}
+
+				return;
+			}, this));
+		};
+		o.cfgvalue = function(section_id) {
```

Comment:
This looks like the default implementation and can be dropped.

Reply:
OK, will do

Reply:
I dropped it according your suggestion.